src/gurt/telemetry.c

```diff
@@ -440,53 +443,91 @@ d_tm_print_timer_snapshot(struct timespec *tms, char *name, int tm_type,
 }
 
 /**
- * Prints the duration \a tms with \a name to the \a stream provided
+ * Prints the duration \a tms with \a stats and \a name to the \a stream
+ * provided
  *
  * \param[in]	tms	Duration timer value
+ * \param[in]	stats	Optional stats
  * \param[in]	name	Duration timer name
  * \param[in]	tm_type	Duration timer type
  * \param[in]	stream	Output stream (stdout, stderr)
  */
 void
-d_tm_print_duration(struct timespec *tms, char *name, int tm_type, FILE *stream)
+d_tm_print_duration(struct timespec *tms, struct d_tm_stats_t *stats,
+		    char *name, int tm_type, FILE *stream)
 {
+	bool printStats;
+
 	if ((tms == NULL) || (name == NULL) || (stream == NULL))
 		return;
 
+	printStats = (stats != NULL) && (stats->k > 0);
+
 	switch (tm_type) {
 	case D_TM_DURATION | D_TM_CLOCK_REALTIME:
-		fprintf(stream, "Duration (realtime): %s = %.9fs\n",
+		fprintf(stream, "Duration (realtime): %s = %.9fs",
 			name, tms->tv_sec + tms->tv_nsec / 1e9);
 		break;
 	case D_TM_DURATION | D_TM_CLOCK_PROCESS_CPUTIME:
-		fprintf(stream, "Duration (process): %s = %.9fs\n",
+		fprintf(stream, "Duration (process): %s = %.9fs",
 			name, tms->tv_sec + tms->tv_nsec / 1e9);
 		break;
 	case D_TM_DURATION | D_TM_CLOCK_THREAD_CPUTIME:
-		fprintf(stream, "Duration (thread): %s = %.9fs\n",
+		fprintf(stream, "Duration (thread): %s = %.9fs",
 			name, tms->tv_sec + tms->tv_nsec / 1e9);
 		break;
 	default:
-		fprintf(stream, "Invalid timer duration type: 0x%x\n",
+		fprintf(stream, "Invalid timer duration type: 0x%x",
 			tm_type & ~D_TM_DURATION);
+		printStats = false;
 		break;
 	}
+
+	if (printStats) {
+		fprintf(stream,
+			" min: %lf max: %lf mean: %lf size: %"
+			PRIu64,
+			stats->dtm_min.min_float,
+			stats->dtm_max.max_float,
+			stats->mean,
+			stats->k);
+		if (stats->k > 2)
+			fprintf(stream, " std dev: %lf", stats->std_dev);
+	}
```

Comment:
I thought it might look unnecessarily cluttered if I wrote this to handle integer and float data types together, so I opted for flattening it out for simplicity.  If there is consensus that we would rather have it reworked in some way as a macro or otherwise, that's fine.  I'm open to suggestions on what that should look like.